crates/edr_provider/src/debug_mine.rs

```diff
@@ -58,6 +58,34 @@ pub struct DebugMineBlockResult<BlockchainErrorT> {
     pub console_log_inputs: Vec<Bytes>,
 }
 
+impl<BlockchainErrorT> DebugMineBlockResult<BlockchainErrorT> {
+    /// Whether the block contains a transaction with the given hash.
+    pub fn has_transaction(&self, transaction_hash: &B256) -> bool {
+        self.block
+            .transactions()
+            .iter()
+            .any(|tx| *tx.hash() == *transaction_hash)
+    }
+
+    /// Returns the index of the transaction with the given hash in the block.
+    pub fn transaction_index(&self, transaction_hash: &B256) -> Option<usize> {
+        self.block
+            .transactions()
+            .iter()
+            .position(|tx| *tx.hash() == *transaction_hash)
+    }
+
+    /// Returns the transaction result of the transaction with the given index.
+    pub fn transaction_result(&self, transaction_index: usize) -> Option<&ExecutionResult> {
+        self.transaction_results.get(transaction_index)
+    }
+
+    /// Returns the transaction trace of the transaction with the given index.
+    pub fn transaction_trace(&self, transaction_index: usize) -> Option<&Trace> {
+        self.transaction_traces.get(transaction_index)
+    }
+}
```

Comment:
These can be combined. See request for change in `data.rs`.

Reply:
Done in [review: use izip instead of index lookup](https://github.com/NomicFoundation/hardhat/pull/5138/commits/1938735f3d6c3c8b825117e73a7904c6bc6e71e2)